### PR TITLE
actually do jit release stuff

### PR DESCRIPTION
**Pull request by aryairani**
Releases now contain `ucm` + three extra things:
* `ui/` - Unison Local gui
* `runtime/` contains a binary that implements the native runtime `run.native`
* `racket/unison.zip` contains a racket library that extends the native runtime to support compiling standalone applications. It can be installed with `raco` to enable ucm's `compile.native` command.

## Implementation notes
* `build-optimized-ucm.yaml` was renamed to `bundle-ucm.yaml`, but includes too many changes to register a rename. It now builds ucm, creates the racket package, builds the `unison-runtime`, and packages all three with the `ui/` as before.
* `pre-release.yaml` can now be run for any branch, to publish binaries for that branch. The tag name it chooses will be derived from the ref name, minus anything before and including a `/`, plus `-build`. 
  e.g. `arya/jit-release` becomes `jit-release-build`.

## Loose ends
- [x] make startup scripts for unix / windows?
- [ ] update homebrew workflow

<!--
**Choose your PR title well:** Your pull request title is what's used to create release notes, so please make it descriptive of the change itself, which may be different from the initial motivation to make the change.

## Overview

What does this change accomplish and why?
i.e. How does it change the user experience?
i.e. What was the old behavior/API and what is the new behavior/API?

Feel free to include "before and after" examples if appropriate. (You can copy/paste screenshots directly into this editor.)

If relevant, which Github issues does it close? (See [closing-issues-using-keywords](https://help.github.com/en/enterprise/2.16/user/github/managing-your-work-on-github/closing-issues-using-keywords).)

## Implementation notes

How does it accomplish it, in broad strokes? i.e. How does it change the Haskell codebase?

## Interesting/controversial decisions

Include anything that you thought twice about, debated, chose arbitrarily, etc.
What could have been done differently, but wasn't? And why?

## Test coverage

Have you included tests (which could be a transcript) for this change, or is it somehow covered by existing tests?

Would you recommend improving the test coverage (either as part of this PR or as a separate issue) or do you think it’s adequate?

If you only tested by hand, because that's all that's practical to do for this change, mention that.

## Loose ends

Link to related issues that address things you didn't get to. Stuff you encountered on the way and decided not to include in this PR.

### CODE REVIEW

File: .github/workflows/pre-release.yaml [hunk 1]
```diff
@@ -33,12 +39,20 @@ jobs:
         with:
           path: /tmp/ucm
 
+      - name: derive release tag
+        run: |
+          if [ -n "${{ github.event.inputs.tag }}" ]; then
+            echo "ref_name=${{ github.event.inputs.tag }}"
+          else
+            echo "ref_name=$(echo ${{ github.ref_name }} | awk -F'/' '{print $NF}')"
```
Comment (ChrisPenner) [inline .github/workflows/pre-release.yaml]: Are you meaning to set env vars here?
At a glance it looks like this just echos it to stdout, but I assume you've tested this somehow?

I didn't see this syntax mentioned in [this guide](https://docs.github.com/en/actions/learn-github-actions/variables#defining-environment-variables-for-a-single-workflow) so am just a bit confused how this works 🤔, I thought you had to write these lines to `$GITHUB_ENV` for it to take effect in future steps.

**Review by ChrisPenner — APPROVED** [review]
Looks good, I see you added an input to pre-release; but I don't see it being triggered manually anywhere, is that just so we can trigger them manually when we want to? May be worth adding a bash script to do that so we don't forget how.

File: .github/workflows/pre-release.yaml (hunk 1 above)
Comment (aryairani) [reply .github/workflows/pre-release.yaml]: Oh yeah, good catch!
This section had a bunch of iteration, a deletion, and then the final version was written by Copilot and I didn't notice the omission.

File: .github/workflows/bundle-ucm.yaml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+                --copy-bins \
+                && break;
+          done
+
+          if [[ ${{runner.os}} = 'Windows' ]]; then
+            ucm=$(stack exec where unison)
+          else
+            ucm=$(stack exec which unison)
+          fi
+          echo ucm="$ucm" > $GITHUB_ENV
```
Comment (aryairani) [inline .github/workflows/bundle-ucm.yaml]:
```suggestion
          echo ucm="$ucm" >> $GITHUB_ENV
```

File: .github/workflows/pre-release.yaml [hunk 3]
```diff
@@ -33,12 +39,20 @@ jobs:
...
+          fi
```
Comment (aryairani) [inline .github/workflows/pre-release.yaml]:
```suggestion
          if [ -n "${{ github.event.inputs.tag }}" ]; then
            ref_name=${{ github.event.inputs.tag }}
          else
            ref_name=$(echo ${{ github.ref_name }} | awk -F'/' '{print $NF}')
          fi
          echo "ref_name=$ref_name" >> $GITHUB_ENV
```

File: scripts/make-release.sh [hunk 4]
```diff
@@ -53,8 +53,8 @@ git fetch origin trunk
 git tag "${tag}" "${target}"
 git push origin "${tag}"
 gh workflow run release --repo unisonweb/unison \
-  --field "version=${version}" \
-  --field "target=${target}"
+  --ref "${target}" \
```
Comment (aryairani) [inline scripts/make-release.sh:56]: Let's see, should this actually be `--ref "${tag}"`? cc @ChrisPenner

**Comment by aryairani** [comment]
> Looks good, I see you added an input to pre-release; but I don't see it being triggered manually anywhere, is that just so we can trigger them manually when we want to? May be worth adding a bash script to do that so we don't forget how.

I think I may just delete that, it's not necessary for triggering manually. I added it to be able to override the tag/url for the pre-release, but now I think it's okay to just be more opinionated about it.

**Pull request merged**